src/main/java/com/redhat/rcm/offliner/Main.java

```diff
@@ -300,6 +308,24 @@ private ArtifactListReader getArtifactListReader( File file )
             try
             {
                 final File target = new File( opts.getDownloads(), path );
+
+                if ( target.exists() )
+                {
+                    if ( null == checksums || checksums.isEmpty() || !checksums.containsKey( path ) )
+                    {
+                        return DownloadResult.avoid( path, true );
+                    }
+
+                    byte[] b = FileUtils.readFileToByteArray( target );
+                    String original = checksums.get( path );
+                    String current = sha256Hex( b );
```

Comment:
Should we do a null check on original here, and skip calculating the file's checksum if it's not present? If I remember correctly, some artifact-list formats (POM, for instance) don't supply a checksum...so if someone used that as an input, this code would read files and compute checksums without any chance of comparing to an original checksum. 

Also, if that's true, we should be careful of NPE in the following 'if' statement.